src/auctionManager.js

```diff
@@ -0,0 +1,452 @@
+    custBidObj.size = custBidObj.getSize();
+
+    utils._each(targeting, function (kvPair) {
+      var key = kvPair.key;
+      var value = kvPair.val;
+
+      if (keyValues[key]) {
+        utils.logWarn('The key: ' + key + ' is getting ovewritten');
+      }
+
+      if (utils.isFn(value)) {
+        try {
+          value = value(custBidObj);
+        } catch (e) {
+          utils.logError('bidmanager', 'ERROR', e);
+        }
+      }
+
+      if (
+        ((typeof bidderSettings.suppressEmptyKeys !== 'undefined' && bidderSettings.suppressEmptyKeys === true) ||
+        key === 'hb_deal') && // hb_deal is suppressed automatically if not set
+        (
+          utils.isEmptyStr(value) ||
+          value === null ||
+          value === undefined
+        )
+      ) {
+        utils.logInfo("suppressing empty key '" + key + "' from adserver targeting");
+      } else {
+        keyValues[key] = value;
+      }
+    });
+
+    return keyValues;
+  }
+
+  function getKeyValueTargetingPairs(bidderCode, custBidObj) {
+    var keyValues = {};
+    var bidder_settings = $$PREBID_GLOBAL$$.bidderSettings;
+
+    // 1) set the keys from "standard" setting or from prebid defaults
+    if (custBidObj && bidder_settings) {
+      // initialize default if not set
+      const standardSettings = getStandardBidderSettings();
+      setKeys(keyValues, standardSettings, custBidObj);
+    }
+
+    // 2) set keys from specific bidder setting override if they exist
+    if (bidderCode && custBidObj && bidder_settings && bidder_settings[bidderCode] && bidder_settings[bidderCode][CONSTANTS.JSON_MAPPING.ADSERVER_TARGETING]) {
+      setKeys(keyValues, bidder_settings[bidderCode], custBidObj);
+      custBidObj.alwaysUseBid = bidder_settings[bidderCode].alwaysUseBid;
+      custBidObj.sendStandardTargeting = bidder_settings[bidderCode].sendStandardTargeting;
+    }
+
+    // 2) set keys from standard setting. NOTE: this API doesn't seem to be in use by any Adapter
+    else if (defaultBidderSettingsMap[bidderCode]) {
+      setKeys(keyValues, defaultBidderSettingsMap[bidderCode], custBidObj);
+      custBidObj.alwaysUseBid = defaultBidderSettingsMap[bidderCode].alwaysUseBid;
+      custBidObj.sendStandardTargeting = defaultBidderSettingsMap[bidderCode].sendStandardTargeting;
+    }
+
+    // set native key value targeting
+    if (custBidObj.native) {
+      Object.keys(custBidObj.native).forEach(asset => {
+        const key = NATIVE_KEYS[asset];
+        const value = custBidObj.native[asset];
+        if (key) { keyValues[key] = value; }
+      });
+    }
+
+    return keyValues;
+  }
+
+  function _createAuction() {
+    const auction = new Auction();
+    _addAuction(auction);
+    return auction;
+  }
+
+  function _getAuction(auctionId) {
+    return _auctions.find(auction => auction.getAuctionId() === auctionId);
+  }
+
+  function _addAuction(auction) {
+    _auctions.push(auction);
+  }
+
+  function Auction() {
```

Comment:
I propose splitting `Auction` out into its own file, for a few reasons.

1. Auctions make sense as a standalone idea.
2. The conceptual dependency between the two is clearly linear, and should never change. AuctionManager can't exist without Auctions, and there's no reason why an individual Auction should ever know about an AuctionManager.
3. Two small files are much easier to maintain than one large one.

There's a (4) too... but it's a little abstract, and I don't think I can describe it in writing. Let me know if you're curious, or unconvinced by the first three and want to chat about it.